include/jemalloc/internal/emap.h

```diff
@@ -293,6 +331,10 @@ emap_alloc_ctx_try_lookup_fast(tsd_t *tsd, emap_t *emap, const void *ptr,
 	if (err) {
 		return true;
 	}
+	/*
+	 * Small allocs using the fastpath can always use index to get the
+	 * usize.  Therefore, do not set alloc_ctx->usize here.
```

Comment:
Let's store the SC_LARGE_MAX + 1 under config_debug. To make sure it's always an invalid size.

Also a question: since we already have alloc_ctx and full_alloc_ctx, do you think we should make another variant that doesn't include the usize, or still have two variants, with only the full_ctx including both edata and usize?

Reply:
If we are going to use `szind` for small allocs and `usize` for the large allocs, we would use a union to store either szind or usize? In that case I would say `alloc_ctx` with cached edata info but no `edata` pointer while `full_alloc_ctx` with `edata` pointer is a reasonable solution.

Reply:
Would we need an additional flag if we use an union, to indicate whether it's szind or size? If that's the case, probably no need to bother.

However maybe worth adding getter for all fields in alloc_ctx like the usize, to inject assertions and disallow arbitrary access to szind / usize. Could be a future PR; up to you.